### PR TITLE
Fix the app crashing when two folders and their ancestors have the same names

### PR DESCRIPTION
**Pull request by aguillon**
Hi!

I think this fixes https://github.com/elementary/code/issues/1440. As mentioned there, a way to reproduce is to open the folders `/usr` and `$HOME/usr`. In this case, the issue is not so much that the two folders and their ancestors have the same names, but that `find_unique_path` does not handle recursion up to the root.

This is my first contribution to a Vala project, ever. In particular, I don't know how to call a code formatter or how to write unit tests (which would make a lot of sense with the present issue). If you show me how, I will gladly update my PR.

### CODE REVIEW

**Comment by jeremypw** [comment]
@aguillon I use `vala-lint` to flag up elementaryOS code style issues but no automatic code formatting at the moment. The CI will fail if there are serious code style errors (as it does currently).

You can get the source code of `vala-lint` from https://github.com/vala-lang/vala-lint.git.

The project does not current implement any unit tests so don't worry about that.  You can look at other projects like Files and Terminal for examples of how it is done.

I have quickly tried out your fix and it seems to be working!  I'll test it a bit more then it should be good to merge.

**Review by jeremypw — CHANGES_REQUESTED** [review]
This fixed a the situation mentioned in the linked issue.  As `get_parent ()` can return `null` I think you need to perform a null check before referencing that parent.

**Comment by aguillon** [comment]
Thank you for `vala-lint`.

> As get_parent () can return null I think you need to perform a null check before referencing that parent.

Thank you, indeed this could happen if we add the root as a project. This should work now.

File: src/Utils.vala [hunk 1]
```diff
@@ -150,10 +150,27 @@ namespace Scratch.Utils {
...
+            if (!f1_parent.has_parent (null)) {
+                f2_parent = f2_parent.get_parent ();
+                break;
+            }
+            if (!f2_parent.has_parent (null)) {
+                f1_parent = f1_parent.get_parent ();
+                break;
+            }
             f1_parent = f1_parent.get_parent ();
             f2_parent = f2_parent.get_parent ();
```
Comment (jeremypw) [inline src/Utils.vala:175]: Should these be null checked too? These parents are referenced in the loop condition.

**Review by jeremypw — CHANGES_REQUESTED** [review]

File: src/Utils.vala (hunk 1 above)
Comment (aguillon) [reply src/Utils.vala:175]: If both previous `has_parent` return `true`, I assume `get_parent` cannot return `null`, no?

Comment (jeremypw) [reply src/Utils.vala:175]: Ah yes, sorry.

**Review by jeremypw — APPROVED** [review]
Lets merge this now!

**Pull request merged**